code/modules/cooking/machinery/cooking_machines/microwave.dm

```diff
@@ -0,0 +1,490 @@
+	else if(dirty > 50)
+		to_chat(user, "It's pretty messy.")
+	else if(dirty > 25)
+		to_chat(user, "It's a bit dirty.")
+
+/*******************
+*   Microwave Menu
+********************/
+
+VUEUI_MONITOR_VARS(/obj/machinery/appliance/cooker/microwave, microwavemonitor)
+	watch_var("operating", "on", CALLBACK(null, .proc/transform_to_boolean, FALSE))
+	watch_var("failed", "failed")
+	watch_var("cook_time", "cook_time")
+	watch_var("start_time", "start_time")
+
+/obj/machinery/appliance/cooker/microwave/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	var/monitordata = ..()
+	if(monitordata)
+		. = data = monitordata
+
+	// init data objects if they aren't already
+	LAZYINITLIST(data["cookingobjs"])
+	LAZYINITLIST(data["cookingreas"])
+
+	// if BYOND lists are smaller than UI, then something (or everything) was removed - wipe the list
+	if(LAZYLEN(contents) < LAZYLEN(data["cookingobjs"]))
+		VUEUI_SET_CHECK_LIST(data["cookingobjs"], list(), ., data)
+	if(LAZYLEN(reagents) < LAZYLEN(data["cookingreas"]))
+		VUEUI_SET_CHECK_LIST(data["cookingreas"], list(), ., data)
+
+	// build the list of objs and reagents
+	if (LAZYLEN(contents))
+		var/list/cook_count = list()
+		for (var/obj/O in contents)
+			cook_count[O.name]++
+		for (var/C in cook_count)
+			VUEUI_SET_CHECK(data["cookingobjs"][C], cook_count[C], ., data)
+	if (LAZYLEN(reagents))
+		for(var/_R in reagents.reagent_volumes)
+			var/singleton/reagent/R = GET_SINGLETON(_R)
+			VUEUI_SET_CHECK(data["cookingreas"][R], REAGENT_VOLUME(reagents, R), ., data)
+
+/obj/machinery/appliance/cooker/microwave/ui_interact(mob/user)
+	var/datum/vueui/ui = SSvueui.get_open_ui(user, src)
+	if (!ui)
+		ui = new(user, src, "machinery-cooking-microwave", 300, 300, capitalize(name))
+	ui.open()
+
+/***********************************
+*   Microwave Menu Handling/Cooking
+************************************/
+
+/obj/machinery/appliance/cooker/microwave/proc/cook()
+	cook_break = FALSE
+	cook_dirty = FALSE
+
+	if(stat & (NOPOWER|BROKEN))
+		return
+
+	if (!reagents && !(locate(/obj) in contents)) //dry run
+		start()
+		return
+
+	recipe = (select_recipe(src, appliance = appliancetype) == recipe)
+
+	if (reagents && prob(50)) // 50% chance a liquid recipe gets messy
+		dirty += Ceiling(reagents.total_volume / 10)
+
+	if (!recipe)
+		failed = TRUE
+		cook_time = update_cook_time()
+		dirty += 5
+		if (prob(max(10, dirty*5)))
+			// It's dirty enough to mess up the microwave
+			cook_dirty = TRUE
+		else if (has_extra_item())
+			// Something's in the microwave that shouldn't be! Time to break!
+			cook_break = TRUE
+	else
+		failed = FALSE
+		cook_time = update_cook_time(round(recipe.time * 2))
+
+	start()
+
+/obj/machinery/appliance/cooker/microwave/proc/update_cook_time(var/ct = 200)
+	RefreshParts()
+	return (ct / cooking_power)
+
+/obj/machinery/appliance/cooker/microwave/process() // What you see here are the remains of proc/wzhzhzh, 2010 - 2019. RIP.
```

Comment:
```suggestion
/obj/machinery/appliance/cooker/microwave/process()
```

Reply:
Not even a memorial for the greatest proc of all time...

Reply:
I don't understand your vendetta against developer comments, Gem.

Don't waste your time requesting changes to remove them, I won't commit them.